tutorials/text_processing/WFST_Tutorial.ipynb

```diff
@@ -0,0 +1,8368 @@
+   "metadata": {
+    "id": "8Ik4PBXafSSB"
+   },
+   "source": [
+    "# NeMo ITN"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "b2fcWKhqYVF5"
+   },
+   "source": [
+    "Following [Google's Kestrel] framework, NeMo deploys two composite WFSTs for text normalization. They are as follows:\n",
+    "1. A *classifier* to label potential tokens by 'semiotic class' (e.g. currency, ordinal number, street address)\n",
+    "2. A *verbalizer* to render a tagged token in conventional written form\n",
+    "\n",
+    "For example, consider the sentence: <<le premier juillet il a mangé trente-cinq pommes>>\n",
+    "\n",
+    "The tokenizer would identify the following tokens:\n",
+    "\n",
+    "`[\"le\" ,\"premier\", \"juillet\", \"il\", \"a\", \"mangé\", \"trente-cinq\", \"pommes\"]`\n",
+    "\n",
+    "and provide each a class token: \n",
+    "\n",
+    "- `tokens { name: \"le\" }`\n",
+    "- `tokens { date { day: \"1\" month: \"juillet\" } } `  \n",
+    "- `tokens { name: \"il\" }` \n",
+    "- `tokens { name: \"a\" }` \n",
+    "- `tokens { name: \"mangé\" }`\n",
+    "- `tokens { cardinal { integer: \"35\" } }` \n",
+    "- `tokens { name: \"pommes\" }`\n",
+    "\n",
+    "These tokens are then passed to a 'verbalizer' WFST, which renders each token in a conventional written form:\n",
+    "\n",
+    "- `tokens { name: \"le\" }` -> `le` \n",
+    "- `tokens { date { day: \"1\" month: \"juillet\" } } ` -> `1ᵉʳ` \n",
+    "- `tokens { name: \"il\" }` -> `juillet`\n",
+    "- `tokens { name: \"il\" }` -> `il` \n",
+    "- `tokens { name: \"a\" }` -> `a`\n",
+    "- `tokens { name: \"mangé\" }` -> `mangé` \n",
+    "- `tokens { cardinal { integer: \"35\" } }` -> `35` \n",
+    "- `tokens { name: \"pommes\" }` -> `pommes`\n",
+    "\n",
+    "and merged into a normalized string:\n",
+    "\n",
+    "`le 1ᵉʳ juillet il a mangé 35 pommes`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "_n-5JExAbvwr"
+   },
+   "source": [
+    "A few things to note: \n",
+    "- NeMo assumes no punctuation (unless explicitly provided in the grammar) and all lower casing to ease integration with upstream ASR.\n",
+    "- The `name` class token is default for any token that does not require normalization. It will be left 'as is.'\n",
+    "- You may note how the tokenizer performed the conversion of `premier` to `1` while the verbalizer normalized `1` -> `1ᵉʳ`. Such decisions are implementation dependent and will vary depending on preference and language. (That is, normalization from `premier` -> `1ᵉʳ` could have been a tokenization step.)\n",
+    "- By default, NeMo ITN will create several permutations of each token to ease normalization. To prevent this and avoid ambiguity in verbalizer input, tokens can be assigned a `preserve_order` attribute to prevent permutation. (e.g. `tokens { date { day: \"1\" month: \"juillet\" preserve_order: true } }` (We will discuss this later in the tutorial.)"
```

Comment:
could you be more speicific here with `permutations of each token`? maybe permution of key values for each semiotic token?